### PR TITLE
Upgrade Query Protocol to Spec V0

### PR DESCRIPTION
**Pull request by hannahhoward**
# Goals

- Use the network abstractions being worked on by @shannonwells in the retrieval implementation
- Upgrade the Query Protocol to follow spec V0

# Implementation

- Make retrieval client/provider dependent on a network protocol implementation rather than directly on a libp2p host
- Add a stubbable mock implementation of the retrieval network (based on @shannonwells work)
- Update query handlers on the provider and client to use query protocol v0 from the spec
- Comment out deal code for the time being which is not v0
- Minor adds / changes to types
- Add tests for retrieval client/provider for queries

### CODE REVIEW

File: retrievalmarket/impl/client.go [hunk 1]
```diff
@@ -110,15 +84,15 @@ func (c *client) Retrieve(ctx context.Context, pieceCID []byte, params retrieval
 
 	go func() {
 		evt := retrievalmarket.ClientEventError
-		converted, err := cid.Cast(pieceCID)
+		//converted, err := cid.Cast(pieceCID)
```
Comment (shannonwells) [inline retrievalmarket/impl/client.go]: Is there a reason to keep this commented-out code or did you mean to delete it?

File: retrievalmarket/impl/client.go [hunk 2]
```diff
@@ -188,7 +163,7 @@ type clientStream struct {
 	verifier   BlockVerifier
 	bs         blockstore.Blockstore
 }
-
+*/
 /* This is the old retrieval code that is NOT spec compliant */
```
Comment (shannonwells) [inline retrievalmarket/impl/client.go]: same re: commented out code

File: retrievalmarket/impl/client.go (hunk 1 above)
Comment (hannahhoward) [reply retrievalmarket/impl/client.go]: Please delete

File: retrievalmarket/impl/client.go (hunk 2 above)
Comment (hannahhoward) [reply retrievalmarket/impl/client.go]: do not delete in this case -- it's the whole old implementation -- need it for reference

File: retrievalmarket/impl/provider.go [hunk 3]
```diff
@@ -159,26 +167,26 @@ type handlerDeal struct {
 }
 
 // TODO: Update for https://github.com/filecoin-project/go-retrieval-market-project/issues/7
-func (p *provider) handleDealStream(stream network.Stream) {
+func (p *provider) HandleDealStream(stream rmnet.RetrievalDealStream) {
 	defer stream.Close()
+	/*
```
[the remark this answers is not in the document]
Comment (hannahhoward) [reply retrievalmarket/impl/provider.go]: no please don't for now. I'm keeping it around as a reference.

[the remark this answers is not in the document]
Comment (shannonwells) [reply retrievalmarket/impl/provider.go]: ok I left that code alone, commenting out handlerDeal struct to please the linter, and rebased w/ master, plus some tidying and that's it.

**Pull request merged**